src/compressed_tensors/compressors/model_compressors/model_compressor.py

```diff
@@ -274,18 +301,24 @@ def __init__(
 
         self.sparsity_compressor = None
         self.quantization_compressor: Optional[
-            Union[BaseQuantizationCompressor, DenseCompressor]
+            Dict[str, Union[BaseQuantizationCompressor, DenseCompressor]]
         ] = None
         # no transform compressor is required
 
         if sparsity_config is not None:
             self.sparsity_compressor = BaseCompressor.load_from_registry(
                 sparsity_config.format, config=sparsity_config
             )
+
         if quantization_config is not None:
-            self.quantization_compressor = BaseCompressor.load_from_registry(
-                quantization_config.format, config=quantization_config
-            )
+            quantization_formats = self._fetch_unique_quantization_formats()
+            self.quantization_compressor = {}
+            for format in quantization_formats:
+                self.quantization_compressor[
+                    format
+                ] = BaseCompressor.load_from_registry(
```

Comment:
Ideally, compression shouldn't require initializing a compressor at all. These compressors don't actually initialize attributes, so it would be nice if they could be reduced to just functions.